src/lib/addons/new-relic.test.ts

```diff
@@ -0,0 +1,166 @@
+        test: '$type toggle',
+    },
+    {
+        partialEvent: {
+            type: FEATURE_ARCHIVED,
+            data: {
+                name: 'some-toggle',
+            },
+        },
+        test: 'FEATURE_ARCHIVED toggle with project info',
+    },
+    {
+        partialEvent: {
+            type: FEATURE_ARCHIVED,
+            project: 'some-project',
+            data: {
+                name: 'some-toggle',
+            },
+        },
+        test: 'FEATURE_ARCHIVED with project info',
+    },
+    {
+        partialEvent: {
+            type: FEATURE_ENVIRONMENT_DISABLED,
+            environment: 'development',
+        },
+        test: 'toggled environment',
+    },
+    {
+        partialEvent: {
+            type: FEATURE_ENVIRONMENT_DISABLED,
+            environment: 'development',
+        },
+        partialParameters: {
+            customHeaders: `{ "MY_CUSTOM_HEADER": "MY_CUSTOM_VALUE" }`,
+        },
+        test: 'customHeaders in headers when calling service',
+    },
+    {
+        partialEvent: {
+            type: FEATURE_ENVIRONMENT_DISABLED,
+            environment: 'development',
+        },
+        partialParameters: {
+            bodyTemplate:
+                '{\n  "eventType": "{{event.type}}",\n  "createdBy": "{{event.createdBy}}"\n}',
+        },
+        test: 'custom body template',
+    },
+] as Array<{
+    partialEvent: Partial<IEvent>;
+    partialParameters?: Partial<INewRelicParameters>;
+    test: String;
+}>)(
+    'Should call New Relic Event API for $test',
+    async ({ partialEvent, partialParameters }) => {
+        const event = {
+            ...defaultEvent,
+            ...partialEvent,
+        };
+
+        const parameters = {
+            ...defaultParameters,
+            ...partialParameters,
+        };
+
+        const handleEvent = makeAddHandleEvent(event, parameters);
+
+        await handleEvent();
+        expect(fetchRetryCalls.length).toBe(1);
+
+        const { url, options } = fetchRetryCalls[0];
+        const jsonBody = JSON.parse(
+            (await asyncGunzip(options.body)).toString(),
+        );
+
+        expect(url).toBe(parameters.url);
+        expect(options.method).toBe('POST');
+        expect(options.headers['Api-Key']).toBe(parameters.licenseKey);
+        expect(options.headers['Content-Type']).toBe('application/json');
+        expect(options.headers['Content-Encoding']).toBe('gzip');
+        expect(options.headers).toMatchSnapshot();
+
+        expect(jsonBody.eventType).toBe('UnleashServiceEvent');
+        expect(jsonBody.unleashEventType).toBe(event.type);
+        expect(jsonBody.featureName).toBe(event.data.name);
+        expect(jsonBody.environment).toBe(event.environment);
+        expect(jsonBody.createdBy).toBe(event.createdBy);
+        expect(jsonBody.createdByUserId).toBe(event.createdByUserId);
+        expect(jsonBody.createdAt).toBe(event.createdAt.getTime());
+
+        expect(options.body).toMatchSnapshot();
```

Comment:
I think you can drop this line, that way the test doesn't rely on confirming that a byte buffer is exactly the same each time. The rest of your test is verification enough

Reply:
Just pushed a new commit removing this line